source/common/router/config_impl.cc

```diff
@@ -480,6 +480,43 @@ void RouteEntryImplBase::finalizePathHeader(Http::HeaderMap& headers,
   headers.Path()->value(path.replace(0, matched_path.size(), rewrite));
 }
 
+absl::string_view RouteEntryImplBase::ProcessRequestHost(const Http::HeaderMap& headers,
+                                                         const absl::string_view& new_scheme,
+                                                         const absl::string_view& new_port) const {
+
+  absl::string_view request_host = headers.Host()->value().c_str();
+  size_t host_end;
+  // Detect if IPv6 URI
+  if (request_host[0] == '[') {
```

Comment:
You can add the ipv6 testcase?

Reply:
Thanks, I'm adding IPv4 and IPv6 testing.